src/app/ffbe/mappers/effects/passives/passive-ailments-cure-after-battle.parser.ts

```diff
@@ -0,0 +1,31 @@
+import {EffectParser} from '../effect-parser';
+import {Skill} from '../../../model/skill.model';
+import {FfbeUtils} from '../../../utils/ffbe-utils';
+
+export class PassiveAilmentsCureAfterBattleParser extends EffectParser {
+  public parse(effect: Array<any>, skill: Skill): string {
+    if (effect.length < 4 || !Array.isArray(effect[3]) || effect[3].length < 8 || effect[3].reduce((a, b) => a + b, 0) < 1) {
+      return 'Effet PassiveAilmentsCureAfterBattleParser inconnu: Mauvaise liste de paramètres';
+    }
+
+    const target = this.getTarget(effect[0], effect[1]);
+
+    const cures = effect[3].map((value, index) => {
```

Comment:
J'aurais mis `.map((value, index) => value === 1 ? this.getAilmentFromId(index + 1) : undefined)`
Ça permet d'explicitement retourner la valeur `undefined` plutôt que de laisser le langage ajouter un `return undefined;` implicite à la fin de ton bloc...